metabase/metabase
# Allow CORS in properties API and mark settings as authenticated

Closes https://github.com/metabase/metabase/issues/47884

### Description

Marks settings that does not need to be public (not i.e. settings required by the setup flow, static and interactive embedding, or user login flow via SSO/LDAP) as authenticated to prevent leaking sensitive information, such as the instance version according to Alex's recommendation in [this thread](https://metaboat.slack.com/archives/C97M2U8M6/p1726072284724699?thread_ts=1726050496.687369&cid=C97M2U8M6).

In addition, it modifies the CORS middleware to always allow CORS for `/api/session/properties` to allow querying public settings from the SDK. This helps the [visual cue in SDK](https://github.com/metabase/metabase/pull/47821) project to be able to query information such as the `embedding-sdk` feature token and `enable-embedding-sdk` settings, to be able to display helpful warning and error UI if the user has a misconfiguration, and they haven't configured CORS yet.

It also marks the `embedding` setting to be public, as that is needed for the visual cue feature in the SDK.

This does NOT allow fetching with cookies, as it does not add the `Access-Control-Allow-Credentials` header, so security-wise it is safe. We won't be able to get the authenticated session properties from the SDK.

### How to verify

- Open a page in your browser, such as `youtube.com`
- Call the `/api/session/properties` endpoint. Try this: `fetch("http://localhost:3000/api/session/properties").then(x => x.json()).then(console.log)`.
- Open the network tab. There should be the origin CORS header now, and we can fetch the public settings now.

### Checklist

- [ ] Add unit test for the endpoint
